### PR TITLE
feat/imx8mp verdin

### PR DESCRIPTION
**Pull request by DavidMCerdeira**
Add support for iMX8MP Verdin Dahlia

Depends on #179

### CODE REVIEW

File: src/platform/drivers/imx_uart/imx_uart.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return;
+}
+
+void uart_putc(volatile struct imx_uart* uart, char c)
+{
+    while (uart->ts & IMX_UART_STAT2_TXFULL)
+        ;
+    uart->txd = c;
+    while (!(uart->stat2 & IMX_UART_STAT2_TXDC))
+        ;
```
Comment (josecm) [inline src/platform/drivers/imx_uart/imx_uart.c:29]:
```suggestion
    while (!(uart->stat2 & IMX_UART_STAT2_TXDC)) { }
```

File: src/platform/imx8mp-verdin/desc.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    .console = {
+        .base = 0x30880000,
+    },
+
+    .arch = {
+
+        .clusters =  {
+            .num = 1,
+            .core_num = (size_t[]) {4}
+        },
```
Comment (josecm) [inline src/platform/imx8mp-verdin/desc.c:27]:
```suggestion

```

No need for a single cluster if there is only one.

Unless mpidrs don't match the expected linearity (i.e., cpu 0 -> mpidr 0x0, cpu 1 -> mpidr 0x1, etc..)

**Review by josecm — CHANGES_REQUESTED** [review]

**Comment by josecm** [comment]
Also @DavidMCerdeira can we add this to the list of supported platforms on the README file?

**Comment by DavidMCerdeira** [comment]
@josecm addressed all you comments

**Comment by josecm** [comment]
Thanks  @DavidMCerdeira. Just need to finish #179 before merging this.

**Review by josecm — DISMISSED** [review]

File: src/platform/imx8mp-verdin/inc/plat/platform.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Bao Project and Contributors. All rights reserved.
+ */
+
+#ifndef __PLAT_PLATFORM_H__
+#define __PLAT_PLATFORM_H__
```
Comment (danielRep) [inline src/platform/imx8mp-verdin/inc/plat/platform.h:7]:
```suggestion
#ifndef PLAT_PLATFORM_H
#define PLAT_PLATFORM_H
```

Let's try to not introduce more identifiers starting with ``__``  which reserved by the C standard.

File: src/platform/imx8mp-verdin/inc/plat/psci.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Bao Project and Contributors. All rights reserved.
+ */
+
+#ifndef __PLAT_PSCI_H__
+#define __PLAT_PSCI_H__
```
Comment (danielRep) [inline src/platform/imx8mp-verdin/inc/plat/psci.h:7]:
```suggestion
#ifndef PLAT_PSCI_H
#define PLAT_PSCI_H
```

Let's try to not introduce more identifiers starting with ``__``  which reserved by the C standard.

File: src/platform/imx8mp-verdin/inc/plat/psci.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#define PSCI_POWER_STATE_LVL_0    0x0000000
+#define PSCI_POWER_STATE_LVL_1    0x1000000
+#define PSCI_POWER_STATE_LVL_2    0x2000000
+#define PSCI_STATE_TYPE_STANDBY   0x00000
+#define PSCI_STATE_TYPE_BIT       (1UL << 16)
+#define PSCI_STATE_TYPE_POWERDOWN PSCI_STATE_TYPE_BIT
+
+#define PLAT_PSCI_POWERDOWN_NOT_SUPPORTED
+
+#endif // __PLAT_PSCI_H__
```
Comment (danielRep) [inline src/platform/imx8mp-verdin/inc/plat/psci.h:18]:
```suggestion
#endif // PLAT_PSCI_H
```

**Review by danielRep — CHANGES_REQUESTED** [review]

File: src/platform/imx8mp-verdin/inc/plat/psci.h (hunk 5 above)
Comment (DavidMCerdeira) [reply src/platform/imx8mp-verdin/inc/plat/psci.h]: Addressed [here](https://github.com/bao-project/bao-hypervisor/compare/32a5a3d596aaeb38279092090d8c0334935055cc..db2d1e34a81c62020cccff58d7bcffeb93269c58) for all discussions

**Review by danielRep — DISMISSED** [review]
Looks good to me!

**Comment by josecm** [comment]
@DavidMCerdeira please run the code formatter.

And don't forget to add `#define PSCI_POWERDOWN_NOT_SUPPORTED 1` to the platform?s header.

**Review by josecm — APPROVED** [review]

**Review by danielRep — APPROVED** [review]

**Pull request merged**